### PR TITLE
Add QuaZip 1.1

### PR DESCRIPTION
**Pull request by ericriff**
Specify library name and version:  **QuaZip/1.1**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericriff** [comment]
I can't get this library to build shared, I get some errors on the `test_package` that I can't figure out.
I've seen @ericLemanissier has helped someone with a similar error, maybe he can give me a hand.
```
/home/eric/.conan/data/quazip/1.1/asd/asd/package/13532c725ecc9b17e2fadf7f5184ae3adc360bd2/lib/libquazip1-qt5.so: undefined reference to `QuaGzipFile::staticMetaObject'
/home/eric/.conan/data/quazip/1.1/asd/asd/package/13532c725ecc9b17e2fadf7f5184ae3adc360bd2/lib/libquazip1-qt5.so: undefined reference to `vtable for QuaZIODevice'
/home/eric/.conan/data/quazip/1.1/asd/asd/package/13532c725ecc9b17e2fadf7f5184ae3adc360bd2/lib/libquazip1-qt5.so: undefined reference to `QuaZipFile::staticMetaObject'
/home/eric/.conan/data/quazip/1.1/asd/asd/package/13532c725ecc9b17e2fadf7f5184ae3adc360bd2/lib/libquazip1-qt5.so: undefined reference to `QuaZIODevice::staticMetaObject'
/home/eric/.conan/data/quazip/1.1/asd/asd/package/13532c725ecc9b17e2fadf7f5184ae3adc360bd2/lib/libquazip1-qt5.so: undefined reference to `vtable for QuaGzipFile'
/home/eric/.conan/data/quazip/1.1/asd/asd/package/13532c725ecc9b17e2fadf7f5184ae3adc360bd2/lib/libquazip1-qt5.so: undefined reference to `vtable for QuaZipFile'
```

**Comment by ericLemanissier** [comment]
I don't have much time for this sorry. The problem could be that moc is not invoked on quazip sources, because the qt cmake files are not packaged.

**Comment by ericriff** [comment]
> I don't have much time for this sorry. The problem could be that moc is not invoked on quazip sources, because the qt cmake files are not packaged.

It looks like you're right.
I've manually built the sources, the log shows
```
-- Build files have been written to: /home/eric/.conan/data/quazip/1.1/asd/asd/source/source_subfolder/build
Scanning dependencies of target QuaZip_autogen
[  5%] Automatic MOC for target QuaZip
[  5%] Built target QuaZip_autogen
Scanning dependencies of target QuaZip
[ 11%] Building CXX object quazip/CMakeFiles/QuaZip.dir/QuaZip_autogen/mocs_compilation.cpp.o
```

Conan logs do not build `QuaZip_autogen` and do not call `auto MOC`
```
-- Build files have been written to: /home/eric/.conan/data/quazip/1.1/asd/asd/build/13532c725ecc9b17e2fadf7f5184ae3adc360bd2
Scanning dependencies of target QuaZip
[  6%] Building C object source_subfolder/quazip/CMakeFiles/QuaZip.dir/unzip.c.o
```

I've overridden the libs generated by conan with the ones I built manually and now the test builds.
I'm not sure what is going on though.

**Comment by ericriff** [comment]
> > I don't have much time for this sorry. The problem could be that moc is not invoked on quazip sources, because the qt cmake files are not packaged.
> 
> It looks like you're right.
> I've manually built the sources, the log shows
> 
> ```
> -- Build files have been written to: /home/eric/.conan/data/quazip/1.1/asd/asd/source/source_subfolder/build
> Scanning dependencies of target QuaZip_autogen
> [  5%] Automatic MOC for target QuaZip
> [  5%] Built target QuaZip_autogen
> Scanning dependencies of target QuaZip
> [ 11%] Building CXX object quazip/CMakeFiles/QuaZip.dir/QuaZip_autogen/mocs_compilation.cpp.o
> ```
> 
> Conan logs do not build `QuaZip_autogen` and do not call `auto MOC`
> 
> ```
> -- Build files have been written to: /home/eric/.conan/data/quazip/1.1/asd/asd/build/13532c725ecc9b17e2fadf7f5184ae3adc360bd2
> Scanning dependencies of target QuaZip
> [  6%] Building C object source_subfolder/quazip/CMakeFiles/QuaZip.dir/unzip.c.o
> ```
> 
> I've overridden the libs generated by conan with the ones I built manually and now the test builds.
> I'm not sure what is going on though.

It is definitely related to the `Qt5` Conan package. I changed the recipe so it uses system's `Qt5` and now `moc` is being called and the target `QuaZip_autogen` is being built :thinking:

File: recipes/quazip/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class QuaZIPConan(ConanFile):
+    name = "quazip"
+    description = "A simple C++ wrapper over Gilles Vollant's ZIP/UNZIP package\
+                    that can be used to access ZIP archives."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/stachenov/quazip"
+    license = "LGPL-2.1-STATIC_LINKING_EXCEPTION"
```
Comment (uilianries) [inline recipes/quazip/all/conanfile.py:10]:
```suggestion
    license = "LGPL-2.1-linking-exception"
```

There is a official short name for it: https://spdx.org/licenses/LGPL-3.0-linking-exception.html

File: recipes/quazip/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            zipFile_path = os.path.abspath(os.path.join(__file__, "..","zipFile.zip"))
+            bin_path = os.path.join("bin", "test_package")
+            self.run(bin_path + " " + zipFile_path, run_environment=True)
```
Comment (uilianries) [inline recipes/quazip/all/test_package/conanfile.py:18]:
```suggestion
		     open("zipfile.zip", "w")
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path + " zipfile.zip", run_environment=True)
```

You don't need an empty file in the repository.

**Comment by ericriff** [comment]
What is going on here?
```
libpng/1.6.37: WARN: Can't find a 'libpng/1.6.37' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Debug, compiler=clang, compiler.version=10, os=Linux
- Options: api_prefix=None, fPIC=True, shared=False, zlib:fPIC=True, zlib:minizip=False, zlib:shared=False
- Dependencies: zlib/1.2.11
- Requirements: zlib/1.Y.Z
- Package ID: 4ada6b9b4eb17cabe936d1e8dc720431e269a9a3

ERROR: Missing prebuilt package for 'libpng/1.6.37', 'qt/5.15.2', 'xkbcommon/1.0.3'
Try to build from sources with '--build=libpng --build=qt --build=xkbcommon'
Use 'conan search <reference> --table table.html'
Or read 'http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package'
```

File: recipes/quazip/all/test_package/conanfile.py (hunk 2 above)
Comment (uilianries) [reply recipes/quazip/all/test_package/conanfile.py:18]: https://github.com/conan-io/conan-center-index/pull/4562/files#diff-9db0766669b7d377d7b1d76b995d978cfe449b16903d8c6229ee688c8d45eb1c It's marked as empty file by Github.

**Comment by uilianries** [comment]
@ericriff That's an internal error. CCI included new compilers so it needs to build all packages again. Unfortunately, not all packages are so far. It should be a temporary error.

@jgsogo @danimtb Could we prioritize those missing packages?

File: recipes/quazip/all/test_package/conanfile.py (hunk 2 above)
Comment (ericriff) [reply recipes/quazip/all/test_package/conanfile.py:18]: I don't know why it says that.
If you follow that link and then click on the 3 dots, select view file and then view raw, you can see it is an actual zip file with 2 txts in it.

Comment (uilianries) [reply recipes/quazip/all/test_package/conanfile.py:18]: understood.

**Comment by jgsogo** [comment]
> @ericriff That's an internal error. CCI included new compilers so it needs to build all packages again. Unfortunately, not all packages are so far. It should be a temporary error.
> 
> @jgsogo @danimtb Could we prioritize those missing packages?

Hi! Sometimes it is not only a missing binary, it is the recipe failing for those new configurations. The best would be to open a PR and see if it builds, or requires modifying the recipe.

**Comment by ericriff** [comment]
So now most of the packages are available, except libpng and qt5 for clang 12 and libpng/1.6.37', 'qt/5.15.2', 'xkbcommon/1.0.3' for clang 10

**Comment by jgsogo** [comment]
Hi! Build for `xkbcommon/1.0.3` is failing for the new compilers: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/TapaholesList/71/summary.json  We need someone to open a PR and provide the patches needed for it to work...

**Comment by ericriff** [comment]
Making progress, the only missing packages now are for qt5 :D

**Review by danimtb — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

File: recipes/quazip/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs=[os.path.join('include', 'QuaZip-Qt5-{}'.format(self.version))]
+        if not self.options.shared:
+            self.cpp_info.defines.append("QUAZIP_STATIC")
+
+        self.cpp_info.filenames["cmake_find_package"] = "QuaZip-Qt5"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "QuaZip-Qt5"
+        self.cpp_info.names["cmake_find_package"] = "QuaZip"
+        self.cpp_info.names["cmake_find_package_multi"] = "QuaZip"
```
Comment (SpaceIm) [inline recipes/quazip/all/conanfile.py:73]: Could be improved in another PR:

```python
    def package_info(self):
        self.cpp_info.libs = tools.collect_libs(self)
        qt_major = tools.Version(self.deps_cpp_info["qt"].version).major
        includedir = os.path.join("include", "QuaZip-Qt{}-{}".format(qt_major, self.version))
        self.cpp_info.includedirs = [
            includedir,
            os.path.join(includedir, "quazip")
        ]
        if not self.options.shared:
            self.cpp_info.defines.append("QUAZIP_STATIC")

        self.cpp_info.filenames["cmake_find_package"] = "QuaZip-Qt{}".format(qt_major)
        self.cpp_info.filenames["cmake_find_package_multi"] = "QuaZip-Qt{}".format(qt_major)
        self.cpp_info.names["cmake_find_package"] = "QuaZip"
        self.cpp_info.names["cmake_find_package_multi"] = "QuaZip"
        quaszip_major = tools.Version(self.version).major
        self.cpp_info.names["pkg_config"] = "quazip{}-qt{}".format(quaszip_major, qt_major)
```

**Pull request merged**